src/modules/alerting.ts

```diff
@@ -122,7 +122,12 @@ class AlertingApi extends BaseApi {
     >(
         title: string,
         message: string,
-        options: ToastOptions<MessageType, Timeout> = {} as ToastOptions<MessageType, Timeout>
+        options: ToastOptions<MessageType, Timeout> = {
+            type: 'info',
+            timeout: 7,
+            showCloseButton: false,
+            markdownMessage: true
+        } as unknown as ToastOptions<MessageType, Timeout>
```

Comment:
So this was an unfortunate cast I had to make to get TS to play nicely. But the most important thing is that we're typesafe from the outside - and we still are here. The heart of the issue is that the consumer of this method can specify a subtype of `ValidMessageType` (MessageType type param) or `number` (Timeout type param) that doesn't intersect with `'info'` or `7` so TS can't assert that this is a safe default.

Reply:
For a minute, I was considering if we should move all these props down to messageParams and overlay the values.  That would let us provid prop defaults even when an options partial was provided by the sdk user.  However, that would be breaking behavior as showCloseButton is evaluated for falsey and undefined would be different than our newly specified default (even if our new defaults are now functionally correct)

I'm assuming that most users are now used to the quirks of this options hash at this point and they're already specifying the props they need.  So, tl;dr, I think we can just keep this as a fully default param and not break the internals too much.

In the future, I think the best path forward might be to provide a more explicit api under different method(s) and deprecate this one; like we discussed.

Reply:
![image](https://user-images.githubusercontent.com/11987207/101119335-186dc100-35b9-11eb-9cd4-f5edfc09e24a.png)

Reply:
Made one last pass at the docs to try to match our behavior.  Take a look and keep me honest.

Reply:
LGTM :shipit: